### PR TITLE
Add ability to customize AppVersionCheck 

### LINKED ISSUE #167: Enhance AppVersionCheck
Suggestion: Enhance the constructor of `OkComputer::AppVersionCheck` to accept `file` and `env` arguments to configure which file path and ENV var to look for. Default to `"REVISION"` and `"SHA"` respectively.

I plan to submit a PR with this enhancement soon.

### PR DESCRIPTION
**Pull request by rmm5t**
## What does it do?

- Adds ability to customize AppVersionCheck with optional `file`, `env`, and `transform` arguments.

## Related Issues

- Closes #167

### CODE REVIEW

**Comment by rmm5t** [comment]
The build failure should be remedied after merging #170 and/or #171

**Comment by rmm5t** [comment]
I'll happily rebase this branch after #170 and #171 are merged.

**Comment by anfleene** [comment]
@rmm5t This is an awesome idea! Thanks for contributing it. I have a few inline concerns that I'll add in review. But the idea is solid and we've been talking about something like this for awhile and never got around to implementing it.

File: lib/ok_computer/built_in_checks/app_version_check.rb [hunk 1]
```diff
@@ -5,6 +5,21 @@ module OkComputer
...
+    attr_accessor :file
+    attr_accessor :env
+    attr_accessor :transform
+
+    # Public: Initialize a check for a backed-up Sidekiq queue
+    #
+    # file - The path of the version file to check
+    # env - The key in ENV to check for a revision SHA
+    # transform - The block to optionally transform the version string
+    def initialize(file = "REVISION", env = "SHA", &transform)
```
Comment (anfleene) [inline lib/ok_computer/built_in_checks/app_version_check.rb:17]: I think I'd rather use [keyword arguments ](https://thoughtbot.com/blog/ruby-2-keyword-arguments) for this. That way the position of the argument or their existence doesn't matter. Like this:
```suggestion
    def initialize(file: "REVISION", env: "SHA", &transform)
```

**Review by anfleene — CHANGES_REQUESTED** [review]

File: lib/ok_computer/built_in_checks/app_version_check.rb (hunk 1 above)
Comment (rmm5t) [reply lib/ok_computer/built_in_checks/app_version_check.rb:17]: @andrewschaar Agreed on the keyword args. I would have actually done that from the get-go, but it didn't look like any existing checks used them. I'll switch.

Comment (rmm5t) [reply lib/ok_computer/built_in_checks/app_version_check.rb]: Switched to keyword args. Let me know if you'd like any other modifications. Thanks.

**Review by anfleene — APPROVED** [review]

**Comment by anfleene** [comment]
Thanks!

**Comment by anfleene** [comment]
[1.18.4](https://rubygems.org/gems/okcomputer/versions/1.18.4) is out with this improvement!

**Pull request merged**